contracts/src/spaces/entitlements/rule/RuleEntitlement.sol

```diff
@@ -166,28 +161,14 @@ contract RuleEntitlement is
     }
 
     Entitlement storage entitlement = entitlementsByRoleId[roleId];
-
-    entitlement.grantedBy = sender;
-    entitlement.grantedTime = currentTime;
-
-    // All checks passed; initialize state variables
-    // Manually copy _checkOperations to checkOperations
-    for (uint256 i = 0; i < checkOperationsLength; i++) {
-      entitlement.data.checkOperations.push(data.checkOperations[i]);
-    }
-
-    for (uint256 i = 0; i < logicalOperationsLength; i++) {
-      entitlement.data.logicalOperations.push(data.logicalOperations[i]);
-    }
-
-    for (uint256 i = 0; i < operationsLength; i++) {
-      entitlement.data.operations.push(data.operations[i]);
-    }
+    entitlement.grantedBy = _msgSender();
+    entitlement.grantedTime = block.timestamp;
+    entitlement.data = data;
```

Comment:
Does this now work because the copy as coming from call data,not memory?

Reply:
Yes but only when the dynamic struct array is nested.
https://github.com/ethereum/solidity/issues/14837

Reply:
Very cool, didn't know this